src/PushNotifications.Ports/TopicSubscriptionPortTracker.cs

```diff
@@ -0,0 +1,25 @@
+﻿using Elders.Cronus;
+using PushNotifications.Contracts;
+using PushNotifications.Contracts.Subscriptions.Events;
+
+namespace PushNotifications.Ports
+{
+    public class TopicSubscriptionPortTracker : IPort,
+      IEventHandler<SubscribedToTopic>,
+      IEventHandler<UnsubscribedFromTopic>
+    {
+        public IPublisher<ICommand> CommandPublisher { get; set; } //bright idea to make it mandatory
```

Comment:
It gets brighter when you use `IGateway`

Reply:
IGateway has a persistent state. Do we need it for Tracker ?

Reply:
`IGateway` is there specifically for situations like this. `IPort` always issues a new command which is not what we want to do here.

Reply:
According to the documentation "Compared to IPort, which can dispatch a command, an IGateway can do the same but it also has a persistent state." Is this outdated and do we consider incrementing a counter in the database to be an external call? Also what do we earn from it having a persistent state.